packages/compass-e2e-tests/installers/mac-dmg.ts

```diff
@@ -0,0 +1,39 @@
+import { existsSync } from 'fs';
+import { assertSpawnSyncResult } from './helpers';
+import type { InstalledAppInfo, Package } from './types';
+import { spawnSync } from 'child_process';
+
+function exec(command: string, args: string[]) {
+  console.log(command, ...args);
+
+  assertSpawnSyncResult(
+    spawnSync(command, args, {
+      encoding: 'utf8',
+      stdio: 'inherit',
+    }),
+    `${command} ${args.join(' ')}`
+  );
+}
+
+export async function installMacDMG(
+  appName: string,
+  { filepath }: Package
+): Promise<InstalledAppInfo> {
+  const fullDestinationPath = `/Applications/${appName}.app`;
+
+  if (existsSync(fullDestinationPath)) {
+    throw new Error(`${fullDestinationPath} already exists`);
+  }
+
+  exec('hdiutil', ['attach', filepath]);
```

Comment:
I'm in two minds about whether this should just be a shell script at this point, but I suppose it is easy to execute a shell script in its place at a later stage. I suppose we'll see what happens once we get to the windows setup .exe and .msi what works best there.

For .tar.gz and .zip files it would likely just be one line anyway.